addons/event/models/event_mail.py

```diff
@@ -76,15 +87,22 @@ class EventMailScheduler(models.Model):
         ('before_event', 'Before the event'),
         ('after_event', 'After the event')],
         string='Trigger ', default="before_event", required=True)
-    template_id = fields.Many2one(
-        'mail.template', string='Email Template',
-        domain=[('model', '=', 'event.registration')], ondelete='restrict',
-        help='This field contains the template of the mail that will be automatically sent')
     scheduled_date = fields.Datetime('Scheduled Sent Mail', compute='_compute_scheduled_date', store=True)
     mail_registration_ids = fields.One2many('event.mail.registration', 'scheduler_id')
     mail_sent = fields.Boolean('Mail Sent on Event', copy=False)
     done = fields.Boolean('Sent', compute='_compute_done', store=True)
 
+    @api.model
+    def _selection_template_model(self):
+        return [('mail.template', 'Mail')]
```

Comment:
@dbeguin It's already before the field `template_ref` :) it's not called `_selection_template_ref`, because it's not a selection field but a reference field (so we just select the model, and not the record)